src/lib/metricsMgr/MetricsManager.h

```diff
@@ -41,28 +94,33 @@
 *    initial map created at constructor time)
 * 3. Destroy method, releasing all the maps in cascade (probably never used, as the
 *    singleton object in CB using this class will be destroyed at the end, but do it
-*    for class compleness)
+*    for class completeness)
 * 4. reset() method implementation (not delete maps, only set metrics to 0)
-* 5. toJson() to be splitted into 3 methods (2 of them private)
+* 5. toJson() to be split into 3 methods (2 of them private)
 * 6. Multi-thread safeness. Probably the same sem-based strategy used in AlarmManager
 *    could be used.
 * 7. Use 'long long' instead of 'int'
-* 8. (Usure) We could need maps for metrics different for int. In that case, implement
-*    it (and the accum method) using templates, to avoid repeat the same implementation
+* 8. (Unsure) We could need maps for metrics different for int. In that case, implement
+*    it (and the add method) using templates, to avoid repeating the same implementation
 *    N times
 */
 class MetricsManager
 {
  private:
-  std::map<std::string, std::map<std::string, std::map<std::string, int>*>* >  metrics;
+  std::map<std::string, std::map<std::string, std::map<std::string, int>*>*>  metrics;
+  sem_t           sem;
+  bool            on;
+  struct timeval  totalTimeInTransaction;
```

Comment:
Right now I am working on the next PR, that will include "serviceTime", and I see now that totalTimeInTransaction cannot be a 'global' value but need to depend on tenant and servicePath.

This (and the method `totalTimeInTransactionAdd`) is being fixed for the next PR.

NTC (in this PR)